mako/lexer.py

```diff
@@ -357,10 +357,12 @@ def match_text(self):
             r"""
                 (.*?)         # anything, followed by:
                 (
-                 (?<=\n)(?=[ \t]*(?=%|\#\#)) # an eval or line-based
+                 ((?<=\n)|^)(?=[ \t]*(?=%|\#\#)) # an eval or line-based
                                              # comment preceded by a
                                              # consumed newline and whitespace
                  |
+                 (?<!%)(?=%%+)
```

Comment:
this can read:

                     (?<!%)(?=%%+)  # consume the first percent sign out of a group of percent signs


so the overall block looks like:

```
        match = self.match(
            r"""
                (.*?)         # anything, followed by:
                (
                 (?<=\n)(?=[ \t]*(?=\#\#)) # an eval or line-based
                                             # comment, preceded by a
                                             # consumed newline and whitespace
                 |
                 (?<!%)(?=%%+)  # consume the first percent sign out of a group of percent signs
                 |
                 (?=\${)      # an expression
                 |
                 (?=</?[%&])  # a substitution or block or call start or end
                              # - don't consume
                 |
                 (\\\r?\n)    # an escaped newline  - throw away
                 |
                 \Z           # end of string
                )""",
            re.X | re.S,
        )
```